src/amoc_api_handler.erl

```diff
@@ -0,0 +1,75 @@
+-module(amoc_api_handler).
+
+-export([init/3]).
+
+-export([rest_init/2,
+         allowed_methods/2,
+         content_types_accepted/2,
+		 content_types_provided/2,
+         from_json/2,
+         process_json/2,
+		 status_response/2]).
+
+-record(state, {action}).
+
+-spec init(tuple(), cowboy:req(), #state{}) -> {upgrade, protocol, cowboy_rest}.
+init({tcp, http}, _Req, _Opts) ->
+    {upgrade, protocol, cowboy_rest}.
+
+-spec rest_init(cowboy:req(), [atom()]) -> {ok, cowboy:req(), #state{}}.
+rest_init(Req, [Action]) ->
+    {ok, Req, #state{action = Action}}.
+
+-spec allowed_methods(cowboy:req(), #state{}) -> {[binary()], cowboy:req(), #state{}}.
+allowed_methods(Req, State) ->
+    {[<<"POST">>, <<"GET">>], Req, State}.
+
+-spec content_types_provided(cowboy:req(), #state{}) -> {[tuple()], cowboy:req(), #state{}}.
+content_types_provided(Req, State) ->
+	{[{<<"application/json">>, status_response}, 
+	  {<<"text/plain">>, status_response},
+	  {<<"text/html">>, status_response}], Req, State}.
+
+-spec content_types_accepted(cowboy:req(), #state{}) -> {[tuple()], cowboy:req(), #state{}}.
+content_types_accepted(Req, State) ->
+    {[{{<<"application">>, <<"json">>, []}, from_json}], Req, State}.
+
+-spec from_json(cowboy:req(), #state{}) -> {true | false, cowboy:req(), #state{}}.
```

Comment:
The spec in line 37 uses a record, please define a type for the record and use that instead.